duckduckgo/duckduckgo-autofill
# Fix autoprompt on slow sites

**Reviewer:** @alistairjcbrown 
**Asana:** https://app.asana.com/0/0/1202727463108108/f

## Description
On some js-heavy sites with `<script defer>` our autoprompt may not fire when the network connection and the site itself are rather slow. This is because when the `load` event fires the page is empty so our initial scan won't report any forms. This change introduces a `shouldAutoprompt` property on the scanner that returns true when called within 1.5 seconds from the object instantiation. This number should be enough to catch most such occasions while also avoiding weird behaviours with super fast sites and super quick users.

## Steps to test
Requires iOS branch [`ema/autoprompt-pixel`](https://github.com/duckduckgo/iOS/pull/1262).

1. On iOS, visit ui.awin.com and save credentials for it (you don't need to have an account, just save fake credentials).
2. Reload the page after saving the credentials.
3. You should now be prompted to use credentials automatically.
